documentation/decisions/adr/0004-api-framework.md

```diff
@@ -0,0 +1,111 @@
+
+### Negative Consequences <!-- OPTIONAL -->
+- Code first paradigm, we should auto generate api documentation in the CI/CD pipeline to ensure it stays up to date with the code
+- This is a relatively new library, so we should ensure code modularity in case we need to swap it out in the future
+
+## Pros and Cons of the Options <!-- OPTIONAL -->
+
+
+### Flask
+
+Flask is a simple, but extensible, micro web framework for python created in 2010 that is easy to learn and build with. While it is capable on its own, it relies on other libraries to add core functionality. It was written to be pythonic, meaning it leverages python's unique features and follows python's principles of being readable and maintainable. Flask depends on the Werkzeug WSGI toolkit, the Jinja template engine, and the Click CLI toolkit.
+
+- **Pros**
+  - Shallow learning curve, great documentation, and pythonic style makes it easy for contributors to support
+  - Flexible and scalable microframework means it can adjust as the needs of the project changes
+  - Designed for backend API services
+- **Cons**
+  - Tech stack can get complicated over time with various libraries
+  - Lack of standardization means more decision making and good code quality is very important
+  - Async operation takes additional planning and work
+
+
+### Flask + Connexion
+
+This deserves its own option because it fundamentally changes the way that we would develop with python and flask. With connexion, you first write your API contract, using the Swagger/OpenAPI standard. Then, the endpoints you defined will be mapped to your python view functions, ensuring that your python code does what your API contract says it does. This makes it rather unique in the landscape of python web frameworks, as most other tools start from your code instead of the other way around.
+
+- **Pros**
+  - Same as Flask
+  - API first means all teams can work with a single contract, instead of having multiple sources of truth
+  - Documentation in OpenAPI or Swagger means it's easier to understand and integrate with other tools
+- **Cons**
+  - Same as Flask
+
+
+### Flask + APIFlask
+
+This deserves its own option because it adds a lot of support to Flask for a more traditional code first api approach, in contrast to connexion. APIFlask is a lightweight Python web API framework based on Flask and marshmallow-code projects. It's easy to use, highly customizable, ORM/ODM-agnostic, and 100% compatible with the Flask ecosystem. The Nava Flask template repo has recently replaced connexion with apiflask, documented in [connexion replacement decision record](https://github.com/navapbc/template-application-flask/blob/main/docs/decisions/0001-connexion-replacement.md).
+
+- **Pros**
+  - Same as Flask
+  - Simplifies boilerplate code necessary for a Flask API by pulling the best features of many libraries
+  - Very well documented
+- **Cons**
+  - Same as Flask
+  - Relatively young project
```

Comment:
Yeah I think this would be my biggest concern with the decision, it looks like it has <1k stars on GH if IIRC, but Flask provides a solid foundation and swapping out RESTful frameworks shouldn't be too hard if we structure the codebase correctly.